File: modules/widgets/lib/rin/widgets/event/CursorDownEvent.cpp
#include "rin/widgets/event/CursorDownEvent.hpp"

CursorDownEvent::CursorDownEvent(const Shared<WidgetSurface>& inSurface, const CursorButton inButton,
        const Vec2<float>& inPosition) : Event(inSurface), button(inButton), position(inPosition)
{
        
}

File: rin.Framework/Widgets/Events/CursorDownEvent.cs
using rin.Framework.Core.Math;
using rin.Framework.Graphics.Windows;
using rin.Framework.Widgets.Graphics;


namespace rin.Framework.Widgets.Events;

public class CursorDownEvent(Surface surface, CursorButton button, Vector2<float> position)
    : Event(surface)
{
    public Vector2<float> Position = position;
    public CursorButton Button = button;
}